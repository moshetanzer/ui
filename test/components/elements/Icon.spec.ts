import { describe, it, expect } from 'vitest'
import { UIcon } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Icon', () => {
  it.each([
    ['with name', { props: { name: 'i-heroicons-academic-cap' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UIcon.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UIcon)
    expect(html).toMatchSnapshot()
  })
})
