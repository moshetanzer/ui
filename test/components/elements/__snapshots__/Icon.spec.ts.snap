// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > renders with name correctly 1`] = `"<span class="i-heroicons-academic-cap"></span>"`;
